html_css/css_math_functions.md

```diff
@@ -0,0 +1,108 @@
+
+### Learning outcomes
+
+* Recognize the basic parts of a CSS function
+* Learn about the `calc()`, `min()`, `max()`, and `clamp()` functions and how to use each one
+
+##### What is a function and how are they used in CSS?
+
+As in other programming languages, functions are reusable pieces of code which perform specific tasks. Functions are passed “arguments” between parentheses, each of which is used by the function in a specific way. Some common examples are:
+
+`color: hsl(230, 100%, 50%);`
+`background: linear-gradient(90deg, blue, red);`
+
+Here, the value of `color` is the function `rgb()`, which accepts arguments in the form of numbers. It processes those numbers to calculate the rgb color corresponding to the three values given. Similarly, the `background` property has a value of `linear-gradient(90deg, blue, red)`, which calculates the angle, starting color, and ending color of a gradient.
+
+Unlike other programming languages you’ll use in TOP, CSS does not allow us to create our own functions. Instead, the language comes bundled with a list of premade functions that will help you solve the most common styling problems. [Here is a complete list](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Functions) of all CSS functions and how they are used.
+
+Besides defining colors, there are several CSS functions that are useful when designing a website’s layout and sizing. These become important when thinking about responsive design.
+
+Let’s  go over a few of these functions: `calc()`, `min()`, `max()`, and `clamp()`.
+
+##### calc()
+The most powerful use cases for calc include:
+
+* Mixing units
+* The ability to nest `calc( calc () - calc () )`
+
+<iframe height="300" style="width: 100%;" scrolling="no" title="calc()" src="https://codepen.io/michaelleojacob/embed/NWgmevp?default-tab=css%2Cresult" frameborder="no" loading="lazy" allowtransparency="true" allowfullscreen="true">
+  See the Pen <a href="https://codepen.io/michaelleojacob/pen/NWgmevp">
+  calc()</a> by Michael Jacob (<a href="https://codepen.io/michaelleojacob">@michaelleojacob</a>)
+  on <a href="https://codepen.io">CodePen</a>.
+</iframe>
+Take a look at how `calc()` is being used here:
+`:root {`
+`--header: 3rem;`
+`--footer: 40px;`
+`--main: calc(100vh - calc(var(--header) + var(--footer)));`
+`}`
+Setting main to equal the outcome of: `100vh - (3rem + 40px)`.
+To put it another way:  `main = 100vh - (header + footer)`.
+`calc()` is handling the math for us even though we are mixing vh, rem and px units.
+Combined with CSS variables, `calc()` can save us from the headache of repeating CSS rules.
+
+I encourage you to take a moment and edit the codepen. Play around with the different units and sizes of the elements by clicking here: <img src="https://imgur.com/a/9iDhtL0" alt="editCodePen">
+
+##### min()
+
+`min()` does an excellent job of helping us create responsive websites. Take a look at this example:
```

Comment:
I like this example... can we add a border to the icon to make extra clear what's happening here?

will also need `box-sizing`

so:
```
#iconHolder {
	width: min(150px, 100%);
	height: min(150px, 100%);
  box-sizing: border-box;
  border: 6px solid blue;
}
```

Reply:
I will touch base with Miggels on how to add this change.